modules/charon-core/src/test/java/org/wso2/charon3/core/protocol/endpoints/GroupResourceManagerTest.java

```diff
@@ -0,0 +1,944 @@
+
+    }
+
+    @DataProvider(name = "dataForTestCreateGroupConflictException")
+    public Object[][] dataToTestCreatGroupConflictException() {
+
+        return new Object[][]{
+                {NEW_GROUP_SCIM_OBJECT_STRING, "id", null}
+        };
+    }
+
+    @Test(dataProvider = "dataForTestCreateGroupConflictException")
+    public void testCreateGroupConflictException(String scimObjectString, String attributes, String excludeAttributes)
+            throws BadRequestException, NotFoundException, CharonException,
+            ConflictException, NotImplementedException {
+
+        mockStatic(AbstractResourceManager.class);
+
+        when(AbstractResourceManager.getResourceEndpointURL(SCIMConstants.USER_ENDPOINT))
+                .thenReturn(SCIM2_GROUP_ENDPOINT);
+        when(AbstractResourceManager.getEncoder()).thenReturn(new JSONEncoder());
+        when(AbstractResourceManager.getDecoder()).thenReturn(new JSONDecoder());
+        when(AbstractResourceManager.encodeSCIMException(any(ConflictException.class)))
+                .thenReturn(getEncodeSCIMExceptionObject(new ConflictException()));
+        when(userManager.createGroup(anyObject(), anyObject())).thenThrow(ConflictException.class);
+
+        SCIMResponse outputScimResponse = groupResourceManager.create(scimObjectString, userManager,
+                attributes, excludeAttributes);
+        Assert.assertEquals(outputScimResponse.getResponseStatus(), ResponseCodeConstants.CODE_CONFLICT);
+
+    }
+
+    @DataProvider(name = "dataForTestCreateGroupNotFoundException")
+    public Object[][] dataToTestCreateGroupNotFoundException() {
+
+        return new Object[][]{
+                {NEW_GROUP_SCIM_OBJECT_STRING, "userName", null}
+        };
+    }
+
+    @Test(dataProvider = "dataForTestCreateGroupNotFoundException")
+    public void testCreateGroupNotFoundException(String scimObjectString, String attributes, String excludeAttributes)
+            throws BadRequestException, NotFoundException, CharonException,
+            ConflictException, NotImplementedException {
+
+        mockStatic(AbstractResourceManager.class);
+
+        when(AbstractResourceManager.getResourceEndpointURL(SCIMConstants.USER_ENDPOINT))
+                .thenReturn(SCIM2_GROUP_ENDPOINT);
+        when(AbstractResourceManager.getEncoder()).thenReturn(new JSONEncoder());
+        when(AbstractResourceManager.getDecoder()).thenReturn(new JSONDecoder());
+        when(AbstractResourceManager.encodeSCIMException(any(NotFoundException.class)))
+                .thenReturn(getEncodeSCIMExceptionObject(new NotFoundException()));
+        when(userManager.createGroup(anyObject(), anyObject())).thenThrow(NotFoundException.class);
+
+        SCIMResponse outputScimResponse = groupResourceManager.create(scimObjectString, userManager,
+                attributes, excludeAttributes);
+        Assert.assertEquals(outputScimResponse.getResponseStatus(), ResponseCodeConstants.CODE_RESOURCE_NOT_FOUND);
+
```

Comment:
Unnecessary new line. and check all similar places.

Reply:
Fixed in commit 71483cb81fc57041180aaa8bc31aa99110211062